TicketingSystem.ApiService/Endpoints/EventEndpoints.cs

```diff
@@ -9,18 +11,26 @@ public class EventEndpoints : IEndpoints
         public void MapEndpoints(IEndpointRouteBuilder app)
         {
             var eventGroup = app.MapGroup("api/events");
-            eventGroup.MapGet("", GetEvents);
-            eventGroup.MapGet("{event_id}/sections/{section_id}/seats", GetSeatsOfSectionOfEvent);
+            eventGroup.MapGet("", GetEvents).CacheOutput(policy => policy.Expire(TimeSpan.FromSeconds(20)).Tag("GetEvents"));
+            eventGroup.MapGet("{eventId}/sections/{sectionId}/seats", GetSeatsOfSectionOfEvent);
         }
 
-        private async Task<Ok<List<TicketsFromEventAndSectionDto>>> GetSeatsOfSectionOfEvent(int event_id, int section_id, IEventService service)
+        private async Task<Ok<List<TicketsFromEventAndSectionDto>>> GetSeatsOfSectionOfEvent(
+            int eventId,
+            int sectionId,
+            IEventService service,
+            IRedisCacheService cache,
+            HttpContext context)
         {
-            var result = await service.GetTicketsOfSectionOfEventAsync(event_id, section_id);
+            var result = await cache.GetSaveAsync(() => service.GetTicketsOfSectionOfEventAsync(eventId, sectionId),
+                $"api/events/{eventId}/sections/{sectionId}/seats", TimeSpan.FromSeconds(6));
```

Comment:
Why do you cache here, in controller and not inside of the event service? Let's imagine a case when you have some other service (class) that calls `IEventService.GetTicketsOfSectionOfEventAsync`. How should it behave then? Should it also cache the result, in the same cache instance as this controller? What cache will be cleared inside of `AddTicketToCartAsync` then?

Reply:
Yeah, it makes more sense to cache in service in case if it has more than 1 caller
Moved caching to the service

But in the example that you've provided, does it matter in what instance cache gets created/cleared? It the same cache everywhere anyways

Reply:
Exactly. All the questions I asked appeared because of how caching is done and are all illuminated when all cache interactions happen inside of the event service.